Fix login form logic

YaronMiro opened the pull request:
#54
![login](https://cloud.githubusercontent.com/assets/3907724/10904853/ed1858ba-821e-11e5-83a5-10102dd454f1.gif)

YaronMiro commented:
@amitaibu 
Ready for review.
Thanks.

amitaibu commented on src/elm/Login.elm:
```diff
@@ -176,15 +176,13 @@ view address model =
 
-    disabledButton =
-      String.isEmpty modelForm.name || String.isEmpty modelForm.pass || model.status == Fetching || model.status == Fetched
+    isStatus = model.status == Fetching || model.status == Fetched
```
isStatus => isFetchStatus

amitaibu commented on src/elm/Login.elm:
```diff
@@ -176,15 +176,13 @@ view address model =
 
     loginText =
-      if disabledButton
+      if isStatus && not (String.isEmpty modelForm.name && String.isEmpty modelForm.pass)
```
make `String.isEmpty modelForm.name && String.isEmpty modelForm.pass` also a var

```elm
isFormEmpty = 
  String.isEmpty modelForm.name && String.isEmpty modelForm.pass
```

amitaibu replied:
so eventual you will have `isFetchStatus || isFormEmpty` - which is very readable.

YaronMiro commented:
@amitaibu 
Ready for review.

I did some more changes:
* The inputs fields are now disabled after clicking on the login button - You were able to change their values  while the request was been made it was weird... ;) 

* Login button color is now gray when disabled - make more scene.

* I removed the variable `loginText` that was holding the "content markup" for the login button and moved all of it to the markup declaration area so know it's more readable and understood.

This is the new video after the changes:
![ezgif com-crop](https://cloud.githubusercontent.com/assets/3907724/10923337/230e5860-8289-11e5-9432-56a821687d98.gif)

amitaibu commented on src/elm/Login.elm:
```diff
@@ -240,9 +239,13 @@ view address model =
-              , disabled disabledButton
+              , disabled (isFetchStatus || isFormEmpty)
+              ]
+              [ span
```
```elm
let
  spinner = 
    i [ class "fa fa-spinner fa-spin" ] []
in
```

```elm
[ span [ hidden <| not isFetchStatus] [ spinner ]
, span [ hidden isFetchStatus ] [ text "Login" ]
]
```

amitaibu commented on src/elm/Login.elm:
```diff
@@ -240,9 +239,13 @@ view address model =
+              , disabled (isFetchStatus || isFormEmpty)
+              ]
+              [ span
+                [ hidden (not (isFetchStatus)) ]
```
`[ hidden (not (isFetchStatus)) ] == [ hidden (not isFetchStatus) ] == [ hidden <| not isFetchStatus ] ==  [ not isFetchStatus |> hidden  ]`

YaronMiro commented:
@amitaibu 
Ready for review.

amitaibu commented:
Merged, thanks!
